bin/api_diff.ml

```diff
@@ -1,55 +1,42 @@
 let tool_name = "api-diff"
 
+type mode = Unwrapped | Wrapped of string | Cmi
+
+let rec mode ~reference ~current ~main_module ~unwrapped =
```

Comment:
This does not need to be recursive. `both_directories` simply needs to be defined first.